Cesium.CodeGen/Contexts/ForScope.cs

```diff
@@ -1,43 +1,71 @@
-using Cesium.CodeGen.Contexts.Meta;
-using Cesium.CodeGen.Ir;
-using Cesium.CodeGen.Ir.Types;
-using Cesium.Core;
-using Mono.Cecil;
-using Mono.Cecil.Cil;
-
-namespace Cesium.CodeGen.Contexts;
-
-internal record ForScope(IEmitScope Parent) : IEmitScope, IDeclarationScope
-{
-    public AssemblyContext AssemblyContext => Parent.AssemblyContext;
-    public ModuleDefinition Module => Parent.Module;
-    public CTypeSystem CTypeSystem => Parent.CTypeSystem;
-    public IReadOnlyDictionary<string, FunctionInfo> Functions => ((IDeclarationScope)Parent).Functions;
-    public TranslationUnitContext Context => Parent.Context;
-    public MethodDefinition Method => Parent.Method;
-    public IReadOnlyDictionary<string, IType> Variables => ((IDeclarationScope)Parent).Variables; // no declarations for `for` now, so pass parent variables
-    public IReadOnlyDictionary<string, IType> GlobalFields => ((IDeclarationScope)Parent).GlobalFields;
-    public void AddVariable(string identifier, IType variable) =>
-        throw new WipException(205, "Variable addition into a for loop scope is not implemented, yet.");
-    public VariableDefinition ResolveVariable(string identifier) => Parent.ResolveVariable(identifier); // no declarations for `for` now, so pass parent variables
-
-    public ParameterDefinition ResolveParameter(string name) => Parent.ResolveParameter(name);
-    public ParameterInfo? GetParameterInfo(string name) => ((IDeclarationScope)Parent).GetParameterInfo(name);
-
-    public Instruction? EndInstruction { get; set; }
-    /// <inheritdoc />
-    public IType ResolveType(IType type) => Context.ResolveType(type);
-    public void AddTypeDefinition(string identifier, IType type) => throw new AssertException("Not supported");
-
-    /// <inheritdoc />
-    public void AddLabel(string identifier)
-    {
-        ((IDeclarationScope)Parent).AddLabel(identifier);
-    }
-
-    /// <inheritdoc />
-    public Instruction ResolveLabel(string label)
-    {
-        return Parent.ResolveLabel(label);
-    }
-}
+using Cesium.CodeGen.Contexts.Meta;
+using Cesium.CodeGen.Ir;
+using Cesium.CodeGen.Ir.Types;
+using Cesium.Core;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Cesium.CodeGen.Contexts;
+
+internal record ForScope(IEmitScope Parent) : IEmitScope, IDeclarationScope
+{
+    public AssemblyContext AssemblyContext => Parent.AssemblyContext;
+    public ModuleDefinition Module => Parent.Module;
+    public CTypeSystem CTypeSystem => Parent.CTypeSystem;
+    public bool TryGetFunctionInfo(string identifier, [NotNullWhen(true)] out FunctionInfo? functionInfo)
+        => ((IDeclarationScope)Parent).TryGetFunctionInfo(identifier, out functionInfo);
+    public TranslationUnitContext Context => Parent.Context;
+    public MethodDefinition Method => Parent.Method;
+    private readonly Dictionary<string, IType> _variables = new();
```

Comment:
I can't find the code that fills this dictionary.

Reply:
I hope transform ForScope into NestedScope in near future. Maybe I should not make code “for future use”

Reply:
In this case, I think that's okay. But maybe we'll want to extract it into a base class.